app/api/src/controllers/taskController.ts

```diff
@@ -112,3 +112,173 @@ export const updateTaskHandler = asyncHandler(
     });
   },
 );
+
+// @desc Creates new planned task list for next day
+// @route POST /task/plan
+// @access Private
+export const planTaskHandler = async (
+  req: AuthRequest,
+  res: Response,
+): Promise<void> => {
+  try {
+    const date = req.query.date;
+
+    if (!date) {
+      res.status(422).json({ error: 'Missing date in query' });
+      return;
+    }
+
+    // Retrieves tasks from prior day
+    const from = moment(date as string)
+      .subtract(1, 'days')
+      .startOf('day');
+    const to = moment(from).add(1, 'day');
+
+    const today = moment(date as string)
+      .startOf('day')
+      .toDate();
+
+    const data = {
+      user_id: req.user_id,
+      date: { $gte: from.toDate(), $lt: to.toDate() },
+    };
+    const tasksToUpdate = await getTasksByDate(data);
+
+    // Adjusted here for empty task list
+    if (!tasksToUpdate || tasksToUpdate.length === 0) {
+      console.log('No tasks for planning');
+      res.status(200).json({
+        message: 'No tasks for planning',
+        updatedTasks: 0,
+        createdTasks: 0,
+      });
+      return;
+    }
+
+    // Filters task list to only modify tasks without status of complete or deleted
+    const tasksUpdateFiltered = tasksToUpdate.filter(
+      (task) => !['Task is complete', 'Task is deleted'].includes(task.status),
+    );
+
+    // Updates the old task to rolled over
+    const updatedTasks = await Promise.all(
```

Comment:
Still not exactly sure if we should be using Promise.all. The more complex we make this, the more I get concerned a random error could occur.